### ISSUE TITLE
Web: Implement Settings Page with Persistent Configuration

### ISSUE THREAD

**Issue by jpequegn** [jpequegn/f1-predict#55]
## Overview
Implement the settings page to allow users to configure application preferences, theme options, API settings, and default behaviors with persistent storage.

## Context
Part of Issue #15 web interface implementation. The settings page stub exists and needs full functionality with preference persistence.

### 1. General Settings

**Display Configuration:**
- [ ] Theme selector (Nebula Dark, Light, High Contrast)
- [ ] Timezone selection (dropdown with major timezones)
- [ ] Unit system (Metric vs Imperial)
- [ ] Language preference (future: for i18n)
- [ ] Date format (ISO, US, EU)

**Application Behavior:**
- [ ] Auto-refresh interval (minutes)
- [ ] Default page on startup
- [ ] Enable notifications toggle
- [ ] Show tooltips toggle

### 2. Prediction Settings

**Model Configuration:**
- [ ] Default ML model (Ensemble, XGBoost, LightGBM, Random Forest)
- [ ] Confidence threshold slider (0.0-1.0)
- [ ] Show feature importance by default toggle
- [ ] Always show explanations toggle
- [ ] Number of predictions to cache

**Display Preferences:**
- [ ] Show podium only vs full results
- [ ] Chart type for confidence (bar, line, scatter)
- [ ] Export format default (CSV, JSON, both)

### 3. Comparison Settings

**Default Filters:**
- [ ] Default season for comparisons
- [ ] Include sprint races toggle
- [ ] Default chart type
- [ ] Number of races to display (5, 10, all)

**Visualization:**
- [ ] Color scheme for drivers
- [ ] Chart animation toggle
- [ ] Show trend lines toggle

### 4. Analytics Settings

**Dashboard Configuration:**
- [ ] Default time period
- [ ] KPIs to display (customizable set)
- [ ] Chart layout (grid, stacked)
- [ ] Auto-refresh enabled/disabled
- [ ] Refresh interval (seconds)

**Data Display:**
- [ ] Standings page size (10, 20, all)
- [ ] Show historical data toggle
- [ ] Include retired drivers toggle

### 5. API Configuration

**Ergast API Settings:**
- [ ] API endpoint URL (default: https://ergast.com/api/f1)
- [ ] Rate limit (requests per second)
- [ ] Request timeout (seconds)
- [ ] Retry attempts on failure
- [ ] Enable API caching toggle
- [ ] Cache duration (minutes)

**Data Collection:**
- [ ] Auto-update data toggle
- [ ] Update frequency (daily, weekly, manual)
- [ ] Data quality threshold
- [ ] Enable data validation toggle

### 6. Advanced Settings

**Performance:**
- [ ] Enable aggressive caching
- [ ] Preload common queries
- [ ] Chart rendering quality (low, medium, high)
- [ ] Maximum concurrent API calls

**Debug:**
- [ ] Enable debug logging toggle
- [ ] Show performance metrics
- [ ] Developer mode toggle
- [ ] Export logs button

**Data Management:**
- [ ] Clear cache button
- [ ] Reset to defaults button
- [ ] Export settings JSON
- [ ] Import settings JSON

### Session State (Temporary)
```python
import streamlit as st

# Initialize settings in session state
if "settings" not in st.session_state:
    st.session_state.settings = load_default_settings()

# Update setting
st.session_state.settings["theme"] = selected_theme
```

### File-Based Persistence
```python
import json
from pathlib import Path

SETTINGS_FILE = Path.home() / ".f1_predict" / "settings.json"

def save_settings(settings):
    """Save settings to JSON file."""
    SETTINGS_FILE.parent.mkdir(parents=True, exist_ok=True)
    with open(SETTINGS_FILE, "w") as f:
        json.dump(settings, f, indent=2)

def load_settings():
    """Load settings from JSON file or return defaults."""
    if SETTINGS_FILE.exists():
        with open(SETTINGS_FILE) as f:
            return json.load(f)
    return get_default_settings()

def get_default_settings():
    """Return default settings configuration."""
    return {
        "theme": "Nebula Dark",
        "timezone": "UTC",
        "units": "metric",
        "default_model": "Ensemble",
        "confidence_threshold": 0.7,
        "enable_explanations": True,
        "api_endpoint": "https://ergast.com/api/f1",
        "rate_limit": 4,
        "cache_duration": 60,
        # ... more defaults
    }
```

### Streamlit Secrets (API Keys)
```toml
# .streamlit/secrets.toml
[api]
ergast_api_key = "optional_key_if_needed"
openai_api_key = "sk-..."
anthropic_api_key = "sk-ant-..."
```

## Implementation Example

```python
def show_settings_page():
    """Display the application settings and preferences."""
    
    st.title("⚙️ Settings")
    
    # Load current settings
    settings = load_settings()
    
    tab1, tab2, tab3, tab4 = st.tabs(["General", "Predictions", "Analytics", "API"])
    
    with tab1:
        st.subheader("General Settings")
        
        theme = st.selectbox(
            "Color Theme",
            options=["Nebula Dark (Default)", "Light", "High Contrast"],
            index=["Nebula Dark (Default)", "Light", "High Contrast"].index(settings.get("theme", "Nebula Dark (Default)"))
        )
        settings["theme"] = theme
        
        timezone = st.selectbox(
            "Timezone",
            options=get_timezones(),
            index=get_timezone_index(settings.get("timezone", "UTC"))
        )
        settings["timezone"] = timezone
        
        units = st.radio(
            "Units",
            options=["Metric (km/h, °C)", "Imperial (mph, °F)"],
            index=0 if settings.get("units") == "metric" else 1,
            horizontal=True
        )
        settings["units"] = "metric" if "Metric" in units else "imperial"
    
    with tab2:
        st.subheader("Prediction Settings")
        
        default_model = st.selectbox(
            "Default Prediction Model",
            options=["Ensemble", "XGBoost", "LightGBM", "Random Forest"],
            index=["Ensemble", "XGBoost", "LightGBM", "Random Forest"].index(settings.get("default_model", "Ensemble"))
        )
        settings["default_model"] = default_model
        
        confidence_threshold = st.slider(
            "Confidence Threshold",
            min_value=0.0,
            max_value=1.0,
            value=settings.get("confidence_threshold", 0.7),
            step=0.05,
            help="Minimum confidence for displaying predictions"
        )
        settings["confidence_threshold"] = confidence_threshold
        
        enable_explanations = st.checkbox(
            "Always show explanations",
            value=settings.get("enable_explanations", True),
            help="Display feature importance and reasoning by default"
        )
        settings["enable_explanations"] = enable_explanations
    
    with tab3:
        st.subheader("API Configuration")
        
        api_endpoint = st.text_input(
            "API Endpoint",
            value=settings.get("api_endpoint", "https://ergast.com/api/f1")
        )
        settings["api_endpoint"] = api_endpoint
        
        rate_limit = st.number_input(
            "Rate Limit (req/sec)",
            min_value=1,
            max_value=10,
            value=settings.get("rate_limit", 4)
        )
        settings["rate_limit"] = rate_limit
        
        cache_duration = st.number_input(
            "Cache Duration (minutes)",
            min_value=5,
            max_value=1440,
            value=settings.get("cache_duration", 60)
        )
        settings["cache_duration"] = cache_duration
    
    # Save button
    col1, col2, col3 = st.columns([1, 1, 3])
    
    with col1:
        if st.button("💾 Save Settings", type="primary", use_container_width=True):
            save_settings(settings)
            st.session_state.settings = settings
            st.success("✅ Settings saved successfully!")
            st.rerun()
    
    with col2:
        if st.button("🔄 Reset to Defaults", use_container_width=True):
            if st.session_state.get("confirm_reset"):
                settings = get_default_settings()
                save_settings(settings)
                st.session_state.settings = settings
                st.session_state.confirm_reset = False
                st.success("✅ Settings reset to defaults!")
                st.rerun()
            else:
                st.session_state.confirm_reset = True
                st.warning("Click again to confirm reset")
    
    # Advanced options
    with st.expander("🔧 Advanced Settings"):
        st.subheader("Data Management")
        
        col1, col2 = st.columns(2)
        with col1:
            if st.button("Clear Cache", use_container_width=True):
                st.cache_data.clear()
                st.cache_resource.clear()
                st.success("✅ Cache cleared!")
        
        with col2:
            if st.button("Export Settings", use_container_width=True):
                settings_json = json.dumps(settings, indent=2)
                st.download_button(
                    label="Download settings.json",
                    data=settings_json,
                    file_name="f1_predict_settings.json",
                    mime="application/json"
                )
```

### Validation
- Required fields highlighted
- Invalid values show error messages
- Prevent invalid state (e.g., rate limit = 0)
- Confirm destructive actions (reset, clear cache)

### Feedback
- Save confirmation message
- Loading indicator during save
- Clear error messages
- Undo option for recent changes (optional)

### Organization
- Logical grouping in tabs
- Clear section headers
- Helpful tooltips
- Consistent layout

### Unit Tests
- [ ] Test settings save/load
- [ ] Test default settings generation
- [ ] Test validation logic
- [ ] Test settings merge (new settings + existing)

### Integration Tests
- [ ] Settings persist across sessions
- [ ] Settings apply to other pages
- [ ] Import/export functionality
- [ ] Reset to defaults works

### Manual Testing
- [ ] All settings save correctly
- [ ] Settings apply immediately or after reload
- [ ] Invalid inputs rejected
- [ ] Mobile layout functional

## Acceptance Criteria

- [ ] Users can configure all settings
- [ ] Settings persist across browser sessions
- [ ] Settings apply to relevant pages
- [ ] Validation prevents invalid configurations
- [ ] Save/reset buttons work correctly
- [ ] Export/import settings functional
- [ ] Clear cache works
- [ ] Mobile responsive layout
- [ ] Help tooltips informative
- [ ] Unit tests pass with 80%+ coverage

## Dependencies

- **Issue #15**: Web interface foundation (completed)
- File system access for settings persistence

## Estimated Effort

**Complexity**: Low-Medium  
**Duration**: 2-3 days  
**Priority**: Low
